hashintel/hash
# Allow creating a property type and inserting it

## 🌟 What is the purpose of this PR?

This makes the create property type modal functional, with basic data type support. Property types are inserted when created. It also adds validation for title and description of a property type. There are some minor visual improvements to the expected values autocomplete (**still not finished**) and added a tooltip on the description. 

This also fixes the `getPropertyType` query.




## 🔗 Related links





- [Asana task](https://app.asana.com/0/1201095311341924/1203172443927341/f) _(internal)_

## 🚫 Blocked by




N/A

## 🔍 What does this change?




- Fixes a bug where the `getPropertyType` query previously actually fetched data types
- Reduced the necessary props for `FontAwesomeIcon` to just the icon part of `IconDefinition` as its the only part we use – enables using pro icons
- Fixed bug in `TextField` where helper text would immediately disappear when cleared, instead of transitioned away
- Made sure `getPropertyType` doesn't permanently cache to ensure that validation works properly (same change was made for `getEntityType`)
- Focus the right field when the create property type modal is opened (and ensure the enter key to open the modal doesn't type a new line in that field)
- Updated design of expected values autocomplete to more closely match design – including adding icons. Not finished.
- Remove null, empty array etc from data types dropdown to match design
- Validate property title uniqueness and description required-ness, and indicate this status
- Create the property type when clicking create – ensuring the property type context value is updated and that the property type is inserted into the entity type

## 📜 Does this require a change to the docs?



N/A

## ⚠️ Known issues



The design of expected values isn't right yet

## 🐾 Next steps



1. Property objects/arrays
2. Design of expected values
3. Editing property types

## 🛡 What tests cover this?



N/A

## ❓ How to test this?



1. Try searching for a property type that doesn't exist in the selector
2. Click the create button
3. Ensure title is pre-populated
4. Enter a description and expected values
5. Click create
6. See that it is inserted into entity type
